tests/TestData/Providers/Stubs/StubMethodsProvider.php

```diff
@@ -27,29 +24,29 @@ public static function allMethodsProvider(): ?Generator
 
     public static function methodsForReturnTypeHintTestsProvider(): ?Generator
     {
-        $filterFunction = fn(PHPClass|PHPInterface $class, PHPMethod $method, $firstSinceVersion) => $class !== null && !$method->isFinal && !$class->isFinal && $firstSinceVersion !== -1 && $firstSinceVersion < 7;
+        $filterFunction = EntitiesFilter::getFilterFunctionForLanguageLevel(7);
         return self::yieldFilteredMethods($filterFunction, StubProblemType::FUNCTION_HAS_RETURN_TYPEHINT);
     }
 
     public static function methodsForNullableReturnTypeHintTestsProvider(): ?Generator
     {
-        $filterFunction = fn(PHPClass|PHPInterface $class, PHPMethod $method, $firstSinceVersion) => $class !== null && !$method->isFinal && !$class->isFinal && $firstSinceVersion !== -1 && $firstSinceVersion < 7.1;
+        $filterFunction = EntitiesFilter::getFilterFunctionForLanguageLevel(7.1);
         return self::yieldFilteredMethods($filterFunction, StubProblemType::HAS_NULLABLE_TYPEHINT);
     }
 
     public static function methodsForUnionReturnTypeHintTestsProvider(): ?Generator
     {
-        $filterFunction = fn(PHPClass|PHPInterface $class, PHPMethod $method, $firstSinceVersion) => $class !== null && !$method->isFinal && !$class->isFinal && $firstSinceVersion !== -1 && $firstSinceVersion < 8;
+        $filterFunction = EntitiesFilter::getFilterFunctionForLanguageLevel(8);
         return self::yieldFilteredMethods($filterFunction, StubProblemType::HAS_UNION_TYPEHINT);
     }
 
-    public static function yieldFilteredMethods(callable $filterFunction, int ...$problemTypes): ?Generator
+    private static function yieldFilteredMethods(callable $filterFunction, int ...$problemTypes): ?Generator
     {
         $coreClassesAndInterfaces = PhpStormStubsSingleton::getPhpStormStubs()->getCoreClasses() +
             PhpStormStubsSingleton::getPhpStormStubs()->getCoreInterfaces();
         foreach (EntitiesFilter::getFiltered($coreClassesAndInterfaces) as $className => $class) {
-            foreach (EntitiesFilter::getFiltered($class->methods, ...$problemTypes) as $methodName => $method) {
-                $firstSinceVersion = StubsTypeHintsTest::getFirstSinceVersion($method); //please vote WI-57343 to fix inspection warning
+            foreach (EntitiesFilter::getFiltered($class->methods, null, ...$problemTypes) as $methodName => $method) {
```

Comment:
Named argument?

Reply:
Spread operator on named argument is forbidden